pyresample/geometry.py

```diff
@@ -2818,6 +2818,8 @@ def get_geostationary_bounding_box_in_lonlats(geos_area, nb_points=50):
     """
     x, y = get_geostationary_bounding_box_in_proj_coords(geos_area, nb_points)
     lons, lats = Proj(geos_area.crs)(x, y, inverse=True)
+    lons[np.isinf(lons)] = np.nan
```

Comment:
Why here we have inf returned by Proj? I remember this function to return the coordinates of the GEO field of view (i.e the circle for FD)  ... I would rather drop Inf/Nan values here !

Reply:
For some reason some of the pixels must end up in space, in which case PROJ returns an infinite value.

Reply:
Isn't the whole point of the `get_geostationary_bounding_box_in_proj_coords` method so that it doesn't return space pixels? Is there a fix we can make to that function to clean this up? Any idea how far the `(x, y)` coordinates producing `inf` are from *not* producing an `inf`? I mean, how close are they to the on-earth disk?

Reply:
I should also say, I don't know if I've tested with the most recent code but I've definitely had cases where polygons with NaNs in the coordinates produced either `NaN` `.area` results or caused infinite loops when trying to do intersections.

Reply:
I'll try to have a look tomorrow.

Reply:
In the test you've added there are 3 locations that result in `inf`. The first, the last, and one in the middle. So my guess is these are two ends of the circle. Let's see how far off the disk they are...

Reply:
Hhhmm the `y` values are definitely not right for a full disk geos area (if that's what this is). The second half are all the same y-position:

```
In [7]: x
Out[7]: 
array([-5.43062255e+06, -5.16482897e+06, -4.39346593e+06, -3.19203985e+06,
       -1.67815466e+06,  3.32529726e-10,  1.67815466e+06,  3.19203985e+06,
        4.39346593e+06,  5.16482897e+06,  5.43062255e+06,  5.16482897e+06,
        4.39346593e+06,  3.19203985e+06,  1.67815466e+06,  3.32529726e-10,
       -1.67815466e+06, -3.19203985e+06, -4.39346593e+06, -5.16482897e+06,
       -5.43062255e+06])

In [8]: y
Out[8]:
array([1393687.2705    , 1672427.79006384, 3181146.69554664,
       4378472.79811701, 5147203.47659387, 5412090.01610633,
       5147203.47659387, 4378472.79811701, 3181146.69554663,
       1672427.79006384, 1393687.2705    , 1393687.2705    ,
       1393687.2705    , 1393687.2705    , 1393687.2705    ,
       1393687.2705    , 1393687.2705    , 1393687.2705    ,
       1393687.2705    , 1393687.2705    , 1393687.2705    ])

```

Reply:
Ok I have to give up. I'm lost in the trigonometry of `get_geostationary_angle_extent` (where is the `1 -` coming from?). I've determined that the x coordinate needs to be moved over 178096 meters before it produces a valid lon/lat. However, if I go the other way and convert lon/lat to x/y, I can do `p(-71.48, 14.83764943645431)` and get `-5252536.339579278` which is about 10 meters further west than I can go in the x/y -> lon/lat direction. Either way we're like 178km from the edge of the projection for some reason.

Reply:
Sorry if I can't have a look ... but I am overwhelmed by other tasks :-(

Reply:
> Hhhmm the `y` values are definitely not right for a full disk geos area (if that's what this is). The second half are all the same y-position:
> 
> Edit: Ah, this is the top half of the SEVIRI image isn't it?

yes, so that's normal

Reply:
Ok, I found the problem. Working on a fix.